edx_sysadmin/views.py

```diff
@@ -34,44 +33,88 @@
     get_course_by_id,
     get_registration_required_extra_fields_with_values,
     is_registration_api_functional,
+    user_has_access_to_users_panel,
+    user_has_access_to_courses_panel,
+    user_has_access_to_git_logs_panel,
+    user_has_access_to_git_import_panel,
 )
+from edx_sysadmin.utils.decorators import user_has_sysadmin_access
+
 
 log = logging.getLogger(__name__)
 
 
+@method_decorator(user_has_sysadmin_access, name="dispatch")
+class SysadminDashboardRedirectionView(RedirectView):
```

Comment:
I don't think we need this view. `user_passes_test` decorator should be use on the sysadmin views and pass `login_url` and `redirect_field_name`.

Reply:
Sorry, it a bit confusing, let me explain my thoughts below to get this comment further:

Whenever a user hits `/sysadmin` this view checks User permissions and redirects the user to the first allowed views to which the user has access like `/sysadmin/users`, `/sysadmin/courses`, `/sysadmin/gitimport` and  `/sysadmin/gitlogs`.

Currently, if a user has access to both `users` and `courses` panel then by default the user will be redirected to `users` panel on hitting the sysadmin base URL `/sysadmin`. And when `user` hits `/courses` URL `intentionally`, he will be redirected to `courses` panel.

Keeping the mentioned flow in mind, we can't put the redirection logic in a decorator and use it on SysadminBaseView because if a user hits `/courses`, `the decorator will pick the first accessible panel and redirect the user to that`.

Reply:
👍